documentation/system-design/book-bazar-openapi.yaml

```diff
@@ -0,0 +1,609 @@
+openapi: 3.0.1
+info:
+  title: Book Bazar
+  description: "[Book Bazar](https://bookbazar.me) API documentation."
+  version: 0.0.0
+externalDocs:
+  description: Check out Book Bazar on GitHub
+  url: https://github.com/calebmech/Book-Bazar
+servers:
+  - url: https://www.bookbazar.me/api
+  - url: https://localhost:4000/api
+tags:
+  - name: Search
+  - name: Post
+  - name: Book
+  - name: Course
+  - name: Auth
+  - name: User
+paths:
+  /search:
```

Comment:
I don't think I understand this endpoint correctly. Is the point to get a list of all posts that match the query, plus the list of all the courses of those posts?

Maybe it's better to separate the searching process in the UI to searching for course, then getting all posts for that course?

Reply:
My thought was that users will either want to finds books for a course, or a specific book. We'd have one unified search bar that the user would type either a course or book title into and we'd give the results like this:

```
┌─────────────────────────────┬────────┐
│ SE 2C03                     │ Search │
├─────────────────────────────┼────────┘
│ SFWRENG 2C03: Data Struc... │         
├─────────────────────────────┤         
│ STATS 2C03: Intro to com... │         
├─────────────────────────────┤         
│ CHEM 2C03: Intro to bioc... │         
├=============================┤         
│ Algorithms - Robert Sedg... │         
├─────────────────────────────┤         
│ Introductory combinatori... │         
├─────────────────────────────┤         
│ Textbook of Biochemistry... │         
└─────────────────────────────┘         
```

The user can then click on any of those entries and they'll all posts for either that course or that book. Or they can click "search" and they'll be taken to a search page with a longer list of courses and books that match their query.

Reply:
Hmm. Okay. I think searching for a postings could be a simpler process if we just add a step where they specify what they want to search by. However, this is more UI stuff that we can figure out later. IMO it would be better to have two separate search endpoints, then we can combine them on the client side however we need. What do you think of this?